src/station_cmd.cpp

```diff
@@ -2200,20 +2200,27 @@ uint8 GetAirportNoiseLevelForTown(const AirportSpec *as, TileIterator &it, TileI
  */
 Town *AirportGetNearestTown(const AirportSpec *as, const TileIterator &it)
 {
+	if (Town::GetNumItems() == 0) return NULL;
+
 	Town *t, *nearest = NULL;
 	uint add = as->size_x + as->size_y - 2; // GetMinimalAirportDistanceToTile can differ from DistanceManhattan by this much
 	uint mindist = UINT_MAX - add; // prevent overflow
```

Comment:
But the `add `business is needed, otherwise CalcClosestTownFromTile can return NULL during the for loop.

Reply:
It will only return NULL if there is no closer town (for that tile) than the current closest. And you already handle the NULL.

Reply:
Perhaps I should assert instead. t should never be NULL inside the for loop. If I don't use the `add`, it will never detect different towns with the same distance.

Reply:
Okay, I understand, you need to find towns with the same distance so that you can find the town with the lowest index, to replicate existing behaviour. Same distance would be catered for by +1, though, and is still unrelated to `GetMinimalAirportDistanceToTile`